From 1f707c57efdeb511e1d9570c3694a7c9327e8dd9 Mon Sep 17 00:00:00 2001
From: Patrick Williams <patrick@stwcx.xyz>
Date: Wed, 4 Sep 2024 06:49:01 -0400
Subject: [PATCH 131/132] ARM: dts: aspeed-g6: add jtag controllers

Summary:

Add jtag controller in aspeed-g6.dtsi.

Pull Request resolved: https://github.com/facebookexternal/openbmc.celestica/pull/1035
GitHub Author: wxiaohua1217 <wxiaohua@celestica.com>

Test Plan:
BMC boot up without error logs.

Reviewers: goldenbug, #openbmc

Reviewed By: goldenbug

Differential Revision: https://phabricator.intern.facebook.com/D27663040

Tags: accept2ship

Signature: 27663040:1617989954:ef64ea4d7a805b0eb4967b7167589aa8afc5562c
---
 arch/arm/boot/dts/aspeed/aspeed-g6.dtsi | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/arch/arm/boot/dts/aspeed/aspeed-g6.dtsi b/arch/arm/boot/dts/aspeed/aspeed-g6.dtsi
index 871ef8899e9d..35beb843c6ba 100644
--- a/arch/arm/boot/dts/aspeed/aspeed-g6.dtsi
+++ b/arch/arm/boot/dts/aspeed/aspeed-g6.dtsi
@@ -380,6 +380,26 @@ rng: hwrng@1e6e2524 {
 				quality = <100>;
 			};
 
+			jtag0: jtag@1e6e4000 {
+				compatible = "aspeed,ast2600-jtag";
+				reg = <0x1e6e4000 0x40 0x1e6e2040 0x8 0x1e6e2418 0x4>;
+				clocks = <&syscon ASPEED_CLK_APB1>;
+				resets = <&syscon ASPEED_RESET_JTAG_MASTER>;
+				interrupts = <27>;
+				status = "disabled";
+			};
+
+			jtag1: jtag@1e6e4100 {
+				compatible = "aspeed,ast2600-jtag";
+				reg = <0x1e6e4100 0x40 0x1e6e2050 0x8 0x1e6e2418 0x4>;
+				clocks = <&syscon ASPEED_CLK_AHB>;
+				resets = <&syscon ASPEED_RESET_JTAG_MASTER2>;
+				interrupts = <GIC_SPI 53 IRQ_TYPE_LEVEL_HIGH>;
+				pinctrl-names = "default";
+				pinctrl-0 = <&pinctrl_jtagm_default>;
+				status = "disabled";
+			};
+
 			gfx: display@1e6e6000 {
 				compatible = "aspeed,ast2600-gfx", "syscon";
 				reg = <0x1e6e6000 0x1000>;
-- 
2.44.2

